data/campaigns/Under_the_Burning_Suns/utils/abilities.cfg

```diff
@@ -73,23 +73,62 @@ _ "This attack uses 6 attacks"#enddef
     [/event]
 #enddef
 
-#define WEAPON_SPECIAL_SONG_VERSE ACTIVE_ON X DISTINGUISHER
+#define STR_VERSE_RESTRICT_1
+_ "This attack can only be used as the first verse of the song."#enddef
+
+#define STR_VERSE_RESTRICT_2
+_ "This attack can only be used as the second verse of the song."#enddef
+
+#define STR_VERSE_RESTRICT_3
+_ "This attack can only be used as the third verse of the song."#enddef
+
+#define STR_VERSE_NAME_1
+_ "first verse"#enddef
+
+#define STR_VERSE_NAME_2
+_ "second verse"#enddef
+
+#define STR_VERSE_NAME_3
+_ "third verse"#enddef
+
+#param MAX is the total number of verses the unit has, needed for id
+#define WEAPON_SPECIAL_SONG_VERSE_BASE MAX
     [disable]
-        id = verse_{DISTINGUISHER}
+        id = verse_{MAX}
+        [filter_self]
+            formula = "self.wml_vars.consecutive_attacks > 1"
+        [/filter_self]
+        active_on = offense
+        name = {STR_VERSE_NAME_1}
+        description = {STR_VERSE_RESTRICT_1}
+    [/disable]
+#enddef
+
+#param X is the verse this attack should be active on
```

Comment:
Why not call it VERSE or VERSE_NUMBER or so? X is not very descriptive.

Btw, nicer to put a space after # so it's easier to distinguish actual comments from preprocessor directives.